framework/test/unit/modules/interoperability/mainchain/commands/sidechain_registration.spec.ts

```diff
@@ -0,0 +1,633 @@
+	STORE_PREFIX_REGISTERED_NETWORK_IDS,
+} from '../../../../../../src/modules/interoperability/constants';
+import {
+	nameSchema,
+	chainIDSchema,
+	sidechainRegParams,
+	chainAccountSchema,
+	channelSchema,
+	validatorsSchema,
+	outboxRootSchema,
+} from '../../../../../../src/modules/interoperability/schema';
+import { SidechainRegistrationParams } from '../../../../../../src/modules/interoperability/types';
+import { VerifyStatus } from '../../../../../../src/node/state_machine';
+import { computeValidatorsHash } from '../../../../../../src/modules/interoperability/utils';
+
+describe('Sidechain registration command', () => {
+	const transactionParams = {
+		name: 'sidechain',
+		genesisBlockID: Buffer.alloc(0),
+		initValidators: [
+			{
+				blsKey: Buffer.from(
+					'3c1e6f29e3434f816cd6697e56cc54bc8d80927bf65a1361b383aa338cd3f63cbf82ce801b752cb32f8ecb3f8cc16835',
+					'hex',
+				),
+				bftWeight: BigInt(10),
+			},
+			{
+				blsKey: Buffer.from(
+					'4c1e6f29e3434f816cd6697e56cc54bc8d80927bf65a1361b383aa338cd3f63cbf82ce801b752cb32f8ecb3f8cc16835',
+					'hex',
+				),
+				bftWeight: BigInt(10),
+			},
+		],
+		certificateThreshold: BigInt(10),
+	};
+	const encodedTransactionParams = codec.encode(sidechainRegParams, transactionParams);
+	const publicKey = getRandomBytes(32);
+	const transaction = new Transaction({
+		moduleID: MODULE_ID_INTEROPERABILITY,
+		commandID: COMMAND_ID_SIDECHAIN_REG,
+		senderPublicKey: publicKey,
+		nonce: BigInt(0),
+		fee: BigInt(100000000),
+		params: encodedTransactionParams,
+		signatures: [publicKey],
+	});
+	const networkIdentifier = Buffer.from(
+		'e48feb88db5b5cf5ad71d93cdcd1d879b6d5ed187a36b0002cc34e0ef9883255',
+		'hex',
+	);
+	const networkID = hash(Buffer.concat([Buffer.alloc(0), transaction.senderAddress]));
+	let sidechainRegistrationCommand: SidechainRegistrationCommand;
+	let db: KVStore;
+	let stateStore: StateStore;
+	let nameSubstore: StateStore;
+	let networkIDSubstore: StateStore;
+
+	beforeEach(() => {
+		sidechainRegistrationCommand = new SidechainRegistrationCommand(
+			MODULE_ID_INTEROPERABILITY,
+			new Map(),
+			new Map(),
+		);
+		db = new InMemoryKVStore() as never;
+		stateStore = new StateStore(db);
+		nameSubstore = stateStore.getStore(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_REGISTERED_NAMES);
+		networkIDSubstore = stateStore.getStore(
+			MODULE_ID_INTEROPERABILITY,
+			STORE_PREFIX_REGISTERED_NETWORK_IDS,
+		);
+	});
+
+	describe('verify', () => {
+		it('should return status OK for valid params', async () => {
+			const context = testing
+				.createTransactionContext({
+					transaction,
+					networkIdentifier,
+				})
+				.createCommandVerifyContext<SidechainRegistrationParams>(sidechainRegParams);
+			const result = await sidechainRegistrationCommand.verify(context);
+			expect(result.status).toBe(VerifyStatus.OK);
+		});
+
+		it('should return error if name is invalid', async () => {
+			const invalidParams = codec.encode(sidechainRegParams, {
```

Comment:
I think if the invalid params are inserted at a later stage, directly into context like `context.params.name` then it'd avoid the overhead of encoding & creating a new `Transaction`. Also since the context creation logic `testing.createTransactionContext` is not different for other tests in `verify` it is possible to define it in a `beforeEach` block. 

Same for all other tests in `verify`